avalon/pipeline.py

```diff
@@ -257,6 +257,10 @@ class InventoryAction(Action):
     order = 0
     hosts = []
 
+    def is_compatible(self, items, **kwargs):
+        """Override function in a custom class"""
+        return True
+
     def process(self, items, **kwargs):
```

Comment:
Instead of `items` aren't they supposed to be `containers`?

Reply:
Eek, yes. And please remove the `**kwargs`, there's no going back from that.

Also, should it not be `container`? Singular? I suppose you can technically multi-select from a GUI, but in that case, should we not rather trigger the action multiple times, and leave intelligence to the GUI?

Reply:
Hm, looks like we've commented on an old commit.

Reply:
We can implement such a logic to the `is_compatible_action`.
I was thinking on something like this:

```
def is_compatible_inventory_action(Action, containers):
    action = Action()
    return any(action.is_compatible(c) for c in container
```

Reply:
Makes more sense (to me) of it too accepting only a single container at a time. At least then it'd align with other "is_compatible" functions.

Reply:
> Makes more sense (to me) of it too accepting only a single container at a time. At least then it'd align with other "is_compatible" functions.

What you suggest is that we create a specific "compatible" function for actions similar to [load_from_representation](https://github.com/getavalon/core/blob/6a2c8edec2fc5cd018d0496d094f8885cde8060c/avalon/pipeline.py#L1141)

I have based the compatibly function on what the [main input](https://github.com/getavalon/core/blob/6a2c8edec2fc5cd018d0496d094f8885cde8060c/avalon/tools/cbsceneinventory/app.py#L43) is which allows us to pass what ever we need to act on the selection. In this case it is `items` / `containers` and based on that data we act. The containers are the input for the action's process method. If there is no container which matches the criteria, the action will not be visible. If there is one that matches, the action needs to be visible.

I think its redundant to make another function in which we loop over x while this function is already simple.